tests/test_link_io_structure.py

```diff
@@ -0,0 +1,243 @@
+from komodo.symlink.sanity_check import (
+    read_link_structure, 
+    equal_links, 
+    assert_root_nodes, 
+    verify_integrity, 
+    sanity_main
+)
+from komodo.symlink.create_links import create_symlinks, symlink_main
+
+import os
+import sys
+import shutil
+import json
+import pytest
+
+def _get_test_root():
+    return os.path.realpath(os.path.dirname(__file__))
+
+def test_read_folder_structure(tmpdir):
+    with tmpdir.as_cwd():
+        os.mkdir('2012.01.rc1')
+        os.mkdir('2012.01.rc2')
+        os.mkdir('bleeding')
+        os.mkdir('2012.01.12')
+
+        os.symlink('2012.01.12', '2012.01')
+        os.symlink('2012.01', 'stable')
+        os.symlink('2012.01', 'testing')
+        os.symlink('2012.01.rc2', 'unstable')
+
+        expected_result = {
+            'root_folder': tmpdir,
+            'root_links' : [
+                'stable',
+                'unstable',
+                'testing'
+            ],
+            'links': {
+                '2012.01': '2012.01.12',
+                'stable': '2012.01',
+                'testing': '2012.01',
+                'unstable': '2012.01.rc2'
+            }
+        }
+
+        output_dict = read_link_structure(os.getcwd())
+
+        assert type(output_dict) == dict
+        assert equal_links(output_dict, expected_result)
+
+
```

Comment:
Please ensure 2 new lines between functions in this file ;)

Reply:
:+1: